### PR TITLE
Ignite basic structure

### PR DESCRIPTION
**Pull request by GantMan**
### basic structure PR 
**comes with 2 command variations**

`./devignite add basic-structure` 
creates a minimal version of the Ignite structure

`./devignite add basic-structure --unholy` 
creates the "KNOWN" ignite structure.  For code crutches.

#### Motivation:
As modules are made, unholy should be subtracted from.  The goal is to subtract unholy down to being a mirror of basic-structure, and thus deleting itself.

#### Pertains to:
Closes: https://github.com/infinitered/ignite/issues/568
Crucial for #567

### CODE REVIEW

File: packages/ignite-basic-structure/templates/App/Config/index.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { Text } from 'react-native'
+import DebugSettings from './DebugSettings'
+import AppConfig from './AppConfig'
+
+export default () => {
+  if (__DEV__) {
+    // If ReactNative's yellow box warnings are too much, it is possible to turn
+    // it off, but the healthier approach is to fix the warnings.  =)
+    console.disableYellowBox = !DebugSettings.yellowBox
```
Comment (skellock) [inline packages/ignite-basic-structure/templates/App/Config/index.js]: Should this be reading from the DebugConfig?

File: packages/ignite-basic-structure/templates/App/Config/index.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  }
+
+  // Allow/disallow font-scaling in app
+  Text.defaultProps.allowFontScaling = AppConfig.allowTextFontScaling
```
Comment (skellock) [inline packages/ignite-basic-structure/templates/App/Config/index.js]: I had no idea you could do this!

File: packages/ignite-basic-structure/templates/App/Containers/Styles/RootContainerStyle.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import {Fonts, Metrics, Colors} from '../../Themes/'
+
+export default StyleSheet.create({
+  applicationView: {
+    flex: 1
+  },
+  container: {
+    flex: 1,
+    justifyContent: 'center',
+    backgroundColor: Colors.background
```
Comment (skellock) [inline packages/ignite-basic-structure/templates/App/Containers/Styles/RootContainerStyle.js]: We don't have `Colors.background` anymore.  Same with fonts and margins.

File: packages/ignite-basic-structure/templates/App/Config/index.js (hunk 1 above)
Comment (GantMan) [inline packages/ignite-basic-structure/templates/App/Config/index.js]: yes will fix

File: packages/ignite-basic-structure/templates/App/Containers/Styles/RootContainerStyle.js (hunk 3 above)
Comment (GantMan) [inline packages/ignite-basic-structure/templates/App/Containers/Styles/RootContainerStyle.js]: 👍

**Pull request merged**